File: map_python/src/Entity/entity.py
from dataclasses import dataclass
from typing import Generic, TypeVar

from .entity_data_state import EntityDataState
from .entity_id import EntityId
from .entity_state import EntityState

T = TypeVar("T")


@dataclass(slots=True)
class Entity(Generic[T]):
    id: EntityId
    name: str
    state: EntityState
    data_state: EntityDataState
    payload: T | None = None

File: map_ts/src/Entity/entity.ts
import type { EntityId } from "./entity-id.js";
import type { EntityState } from "./entity-state.js";
import type { EntityDataState } from "./entity-data-state.js";

export type Entity<
  TPayload = unknown
> = {
  id: EntityId;
  name: string;

  state: EntityState;
  dataState: EntityDataState;

  payload?: TPayload;
};